keps/sig-node/4569-cgroup-v1-maintenance-mode/README.md

```diff
@@ -0,0 +1,646 @@
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Move cgroup v1 support in Kubernetes into maintenance mode, aligning with the industry's move towards cgroup v2 as the default for Linux kernel resource management and isolation.
+
+## Motivation
+
+cgroup v2 offers significant improvements over cgroup v1 in terms of feature set, uniformity of interface, and scalability. With major Linux distributions and container runtimes increasingly supporting cgroup v2, Kubernetes aims to encourage the adoption of this newer, more secure, and more performant technology. By transitioning cgroup v1 support to maintenance mode, Kubernetes can ensure stability for existing deployments while simultaneously promoting the gradual adoption of cgroup v2.
+
+### Goals
```

Comment:
May I suggest to add a new env variable (or a feature gate) that needs to be set to enable cgroupv1? I think it will be very useful for spreading awareness among the people deploying k8s. Also it will help us with tests - we can check that we are not adding new cgroupv1 test runs in our test grid easier, without the need to review the configuration files for jobs

Reply:
We can change the defaulting in different places, too. 

For example, in some future minor release, imagine that:
- kubeadm defaults to explicitly disabling cgroupv1 support
- kubelet defaults to having cgroup support enabled, but warns if that support actually gets used.
  You can still disable the cgroupv1 support and then you get an error if the kubelet can't find v2 cgroup support from the init process / whatever.

Reply:
>May I suggest to add a new env variable (or a feature gate) that needs to be set to enable cgroupv1? I think it will be very useful for spreading awareness among the people deploying k8s. Also it will help us with tests - we can check that we are not adding new cgroupv1 test runs in our test grid easier, without the need to review the configuration files for jobs

@SergeyKanzhelev  This was the original plan, but the feedback on this KEP has been not to disrupt existing cgroup v1 support. If we put any barrier to enable cgroup v1, it will disrupt their workflow as they will have to perform this additional step to keep using v1.

Reply:
I don't think we should do this. _maybe_ one release before actually deleting support instead of only announcing it, but we aren't even currently planning to drop support entirely within the scope of this KEP now anyhow?

Something to consider for a future KEP to drop support, I think?